src/mlpack/methods/ann/layer/batch_norm.hpp

```diff
@@ -44,39 +47,81 @@ namespace ann /** Artificial Neural Network. */ {
  * }
  * @endcode
  *
- * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
- *         arma::sp_mat or arma::cube).
- * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
- *         arma::sp_mat or arma::cube).
+ * @tparam MatType Matrix representation to accept as input and use for
+ *         computation.
  */
-template <
-  typename InputType = arma::mat,
-  typename OutputType = arma::mat
->
-class BatchNorm : public Layer<InputType, OutputType>
+template <typename MatType = arma::mat>
+class BatchNormType : public Layer<MatType>
 {
  public:
-  //! Create the BatchNorm object.
-  BatchNorm();
+  /**
+   * Create the BatchNorm object. 
+   * 
+   * This sets the minimum and maximum axis for batch normalization to 2; so, 
+   * e.g., we apply batch normalization to the first 3 dimensions with number 
+   * of channels equal to 3rd dimension, and higher dimensions are left 
+   * untouched.  As an example, if we have a 3-dimensional input (call the 
+   * three dimensions rows, columns and slices), and `minAxis` & `maxAxis` is 
+   * 2, then we apply the same normalization across different slices.
```

Comment:
```suggestion
   *
   * With batch normalization, the same exact normalization is applied to every
   * element in an individual channel.  To control what axes normalization is
   * applied to, set the `minAxis` and `maxAxis` parameters.
   *
   * The last axis of the input data will be chosen as channels.  So, if the
   * input is 3-dimensional or higher, this constructor will set the minimum
   * and maximum axes to 2; this will take only the 3rd axis of the input as
   * channels. If the input is 1-dimensional, then the minimum and maximum axis
   * will be 0, and thus every element of the input will have a different
   * normalization applied to it.
   *
   * As an example, if we have a 3-dimensional input (call the 
   * three dimensions rows, columns and slices), and `minAxis` & `maxAxis` is 
   * 2, then we apply the same normalization across different slices.
```
I tried to make this match the wording of the other constructor a little better (so it talks about channels now).  Hopefully this helps. :+1: